src/main/scala/tech/cryptonomic/conseil/util/CollectionOps.scala

```diff
@@ -37,5 +39,13 @@ object CollectionOps {
     def onBounds(f: (T, T) => R): Option[R] = applyOnBounds(list)(f)
   }
 
+  implicit class ExtendedFutureWithOption[T](val value: Future[Option[List[T]]]) {
+    def embeddedMap(function: T => Option[T])(implicit apiExecutionContext: ExecutionContext): Future[Option[List[T]]] =
```

Comment:
minor: the name to me sounds like a data instance, possibly `mapNested`/`mapNested` could make it more evident?